### ISSUE TITLE
Docker compose environment variables not taking affect after initial setup

### ISSUE THREAD

**Issue by volt3** [nextcloud/docker#189]
Hello!

TLDR: `MYSQL_HOST` does not modify `dbhost` inside config.php after initial setup. 

I modified the basic docker compose file suggested in the README by adding the following:
```
services:
  db:
    ports:
      - 8081:3306
```

Now that I understand compose creates a network between the services and I do not need to expose the port to host, I removed the above and understand that the following is sufficient. 
```
services: 
  app:
    links:
      - db
```
However this did not modify `dbhost` inside /html/config.php as expected.

I added the following to the docker-compose.yml: 
```
services:
  app:
    environment:
      MYSQL_DATABASE=XXXX
      MYSQL_USER=XXXX
      MYSQL_PASSWORD=XXXX
      MYSQL_HOST=db
```
And this did not modify `dbhost`, it was still the same value as initial setup. 

I fixed my problem by modifying `dbhost` to just `db`. 

**Problems**
* As the docker-compose file does not seem to have precedence(after the initial setup), if I change the service name from `db` to `nextcloudDataBase`, and `links: ` to `nextcloudDataBase` it breaks the deployment although docker-compose networks will be able to resolve `http://nextcloudDataBase`  
* I naturally expect/assume/define that when I set `MYSQL_HOST` within docker-compose it will look for the correct host after a container has been removed and recreated.

**Question**
1) Although changing the `dbhost` fixed my network issue, is `MYSQL_HOST` intended to modify any other variable/aspect of the system? 

Thank you for your time.

**Comment by SnowMB** [comment]
Hello,

I hope I can clarify things a bit. As you found out, the environment variables are only used by the initial setup. This is intended and by design. It might not be the best solution, but it was the easier one. I see that many other services handle things a little differently, so we might have a chance to improve here.

For your problem there are two solutions:

If you are trying to get a new installation of nextcloud running and have no data that could be lost, you can just start from scratch. But be careful, stopping and recreating containers will preserve the data! To actually delete the screwed up configuration you have to explicitly tell docker to remove the volumes where your data and configuration is stored. You can achieve this by adding the `-v` flag to `docker-compose down -v`. Or you can manually remove the volumes by using the following commands:

```
docker volume ls 
docker volume rm <NAME>
```

If you want to keep your data, the only way to fix your setting is editing your config.php manually. Check out the [nextcloud docs](https://docs.nextcloud.com/server/12/admin_manual/configuration_server/config_sample_php_parameters.html) for the correct parameters.


Sidenote:
Using `link:` in the docker-compose file is not necessary and deprecated in newer version of the [compose file reference](https://docs.docker.com/compose/compose-file/). You can just use the service name (`db` in this case). Just use `depends_on:` to configure the starting order.

**Comment by SnowMB** [comment]
Sorry for closing...

**Comment by volt3** [comment]
Thank you for your time. 

I have done the changes I needed and am very happy with the current install.

**Comment by evertramos** [comment]
Hello @volt3,

Just to answer your final question the `MYSQL_HOST` only changes the `./config/config.php` as you probably already know by now.

@SnowMB think you can close this issue.

**Comment by SnowMB** [comment]
Yeah, think so to. Reopen if another related question occurs.
